apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/RowEditor/ForeignRowSelector/ForeignRowSelector.tsx

```diff
@@ -79,6 +82,7 @@ const ForeignRowSelector = ({
       filters,
       page,
       limit: rowsPerPage,
+      impersonatedRole: roleImpersonationState.role,
```

Comment:
just a nit - why not use `getImpersonatedRole()` instead of declaring `roleImpersonationState`?